addons/l10n_ro_efactura/models/ciusro_document.py

```diff
@@ -0,0 +1,187 @@
+    state = fields.Selection(
+        selection=[
+            ('invoice_sending', 'Sending'),
+            ('invoice_sending_failed', 'Error'),
+            ('invoice_sent', 'Sent'),
+        ],
+        string='E-Factura Status',
+        required=True,
+    )
+    datetime = fields.Datetime(default=fields.Datetime.now, required=True)
+    attachment_id = fields.Many2one(comodel_name='ir.attachment')
+    message = fields.Char()
+    key_loading = fields.Char()         # To be used to fetch the status of previously sent XML
+    key_signature = fields.Char()       # Received from a successful response: to be saved for government purposes
+    key_certificate = fields.Char()     # Received from a successful response: to be saved for government purposes
+
+    def unlink(self):
+        """ This method make sure that the created attachment (if it exists) is also deleted """
+        self.attachment_id.sudo().unlink()
+        return super().unlink()
+
+    @api.model
+    def _make_efactura_request(self, company, endpoint, method, params, data=None, session=None) -> dict[str, str | bytes]:
+        """ Make an API request to the Romanian SPV, handle the response, and return a `result` dictionary
+            :param company: `res.company` object containing l10n_ro_edi_test_env, l10n_ro_edi_access_token
+            :param endpoint: 'upload' (for sending) |
+                             'stareMesaj' (for fetching status) |
+                             'descarcare' (for downloading answer)
+            :param method: 'post' (for 'upload') |
+                           'get' (for 'stareMesaj'|'descarcare')
+            :param params: Dictionary of query parameters
+            :param data: xml data for 'upload' request
+            :return: dictionary of {'error': <str>} or {'content': <response.content>} from E-Factura
+        """
+        send_mode = 'test' if company.l10n_ro_edi_test_env else 'prod'
+        url = f"https://api.anaf.ro/{send_mode}/FCTEL/rest/{endpoint}"
+        headers = {'Content-Type': 'application/xml',
+                   'Authorization': f'Bearer {company.l10n_ro_edi_access_token}'}
+
+        try:
+            requester = session or requests
+            response = requester.request(method=method, url=url, params=params, data=data, headers=headers, timeout=10)
+        except requests.HTTPError as e:
+            return {'error': str(e)}
+        if response.status_code == 400:
+            error_json = response.json()
+            return {'error': error_json['message']}
+        if response.status_code == 403:
+            return {'error': _('Access token is forbidden.')}
+        if response.status_code == 204:
+            return {'error': _('You reached the limit of requests. Please try again later.')}
+
+        return {'content': response.content}
+
+    @api.model
+    def _request_ciusro_send_invoice(self, company, xml_data, move_type='out_invoice'):
```

Comment:
Okay, I changed the `_make_efactura_request` to be a normal function, but I will keep the `_request_ciusro...` a model function in l10n_ro_edi.document, because we might want to call it from multiple places in the future, not just account.move